agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -1023,6 +1023,7 @@ bool slave_thread::handle_cmdu_control_message(
             return false;
         }
         configuration_stop_on_failure_attempts = request_in->attempts();
+        stop_on_failure_attempts               = configuration_stop_on_failure_attempts;
```

Comment:
Why do we need both `configuration_stop_on_failure_attempts` and `stop_on_failure_attempts`?

Reply:
Because stop_on_failure_attempts is decremented on every failure, but reset to its original value on success.